// vue-tutorial/src/factory/highchart.js

import ChartMaker from '../factory/gbasechart';

import * as DataManager from '../highchart/highchartData';
import BarChart from '../highchart/barChart';
import ColumnChart from '../highchart/columnChart';
import LineChart from '../highchart/lineChart';
import AreaChart from '../highchart/areaChart';
import PieChart from '../highchart/pieChart';
import BubbleChart from '../highchart/bubbleChart';
import GenreMaker from '../highchart/genreMaker';

var HighChart = class HighChart extends ChartMaker {

    draw = function(state){
      try{
        //$("#container").html("");
        this.propsToVO(state.props);
        this.chart = this.createChartByGenre(state.chartInfo);
        this.setChartProperties();
        //this.chart.container('container');
        function chart3(){
          var dataType = state.chartInfo.subChart.highdataType;
          var data = DataManager.getData(dataType);
          data.title.text = state.props.title.value;
          if(state.chartInfo.chart.genre == "line")
          {
            data.chart.type="line";
            if(dataType=='spline' || dataType=='series_spline')
            {
              data.chart.type="spline";
            }
          }

          if(state.chartInfo.chart.genre == "bar")
          {
            data.chart.type="bar";
            if(dataType=='percent_stacked')
            {
              data.plotOptions={bar:{stacking:'percent'}};
            }

          }

          // if(state.chartInfo.chart.genre == "pie")
          // {
          //   data.chart.type="pie";
          // }
          //
          // if(state.chartInfo.chart.genre == "area")
          // {
          //   data.chart.type="area";
          // }

          var chart = new Highcharts.Chart(data);
        }
        $(function () {
            chart3();
        });
        this.chart.draw();
      }
      catch(e)
      {
        document.getElementById("container").innerHTML = "图形维护中...3秒后自动跳转到可用图形";
        setTimeout(load,2000);
        function load(){
          location.reload();
        }



      }

    }

    setTitle = function(title){
        this.title = title;
        this.chart.title(title);
    }

    JSONToVO = function(json){
       this.title = json.chart.title.text;
    }

    propsToVO = function(props){
        this.title = props.title.value;
    }

    createChartByGenre = function(chartInfo){
        var genre = chartInfo.chart.genre;
        var subGenre = chartInfo.subChart.genre;
        var dataType = chartInfo.subChart.highdataType;
        //var data = DataManager.getData(dataType);
        //return GenreMaker.factory(genre).getChart(subGenre,data);
        return GenreMaker.factory(genre).getChart(subGenre,"");
    }

    setChartProperties = function(props){
        this.setTitle(this.title);
    }
};

ChartMaker.HighChart = HighChart;

export var HighChart;
